cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/sync/class-push-sync.php

```diff
@@ -570,6 +568,13 @@ public function push_attachments( $attachments ) {
 						$args['context'] = $upload['options']['context'];
 					}
 					$result = $this->plugin->components['connect']->api->explicit( $args );
+				} elseif ( 'rename' === $sync_type ) {
+					// Rename an asset.
+					$args   = array(
+						'from_public_id' => $this->plugin->components['media']->get_post_meta( $attachment->ID, Sync::META_KEYS['public_id'] ),
+						'to_public_id'   => $upload['public_id'],
+					);
+					$result = $this->plugin->components['connect']->api->{$upload['options']['resource_type']}( 'rename', 'POST', $args );
```

Comment:
Not sure what's going on here. This abstraction is very hard to understand without dumping out the data.

Frankly, this method is _massive_ and could use some refactoring. Very hard to understand what's going on overall.